### PR TITLE
Fixes #72

### LINKED ISSUE #72: Non obvious behavior when passing in both stack_info and exc_info
* PySeq version: latest
* Python version: 3.11
* Operating System:  Linux

### Description

The behavior of the logger when passed in both stack_info, exc_info and (or) exc_text is super counterintuitive and is not documented. It needs to be documented. I'll do it.

### Also
I feel that SeqLogHandler should check for _support_stack_info and the remaining features every time it needs them, instead inside of the constructor and just once, since later changes won't work, and it's a cheap operation.

### PR DESCRIPTION
**Pull request by piotrmaslanka**

### CODE REVIEW

File: seqlog/consumer.py [hunk 1]
```diff
@@ -58,10 +58,10 @@ def flush(self):
 
             self._cancel_auto_flush()
 
-            current_batch = self.current_batch[:]
+            current_batch = [*self.current_batch]
```
Comment (piotrmaslanka) [inline seqlog/consumer.py:61]: Turns out it's the fastest way to shallow copy short lists.

File: seqlog/consumer.py [hunk 2]
```diff
@@ -58,10 +58,10 @@ def flush(self):
...
             self.current_batch.clear()
 
-            if len(current_batch) == 0:
+            if not current_batch:
```
Comment (piotrmaslanka) [inline seqlog/consumer.py:64]: We already know that it's a list.

File: docs/usage.rst [hunk 3]
```diff
@@ -1,213 +1,233 @@
...
+
+Passing in exceptions
+---------------------
+
+There's a couple of ways you can pass in exception information. The only field sent to Seq will be called Exception and it
+may come from a couple of places, in order:
+
+1. A previously rendered exception was cached
+
+2. If FeatureFlag.STACK_INFO is enabled, :code:`record` is set and :code:`record` is not set, it
```
Comment (tintoy) [inline docs/usage.rst:223]: Just checking - is this a typo? Is one of these meant to say `stack_info`?

(I get that it was already there)

File: docs/usage.rst [hunk 4]
```diff
@@ -1,213 +1,233 @@
...
+   will be taken from :code:`stack_info`
+
+3. If :code:`exc_info` is a tuple then
+    a. If it's first element is None, FeatureFlag.STACK_INFO is enabled and :code:`record.stack_info` is available,
+       :code:`stack_info` will be used to construct the message
+    b. Else formatted :code:`exc_info` will be used
+
+4. If only :code:`exc_info` is given, and it's an Exception, then the stack trace will be attached.
+
+So if you provide both :code:`exc_info` and :code:`stack_info` the code will behave in a weird way.
```
Comment (tintoy) [inline docs/usage.rst:233]: Can I suggest “the code will behave in a weird way” -> “behaviour is undefined”?

File: seqlog/consumer.py (hunk 1 above)
Comment (tintoy) [reply seqlog/consumer.py:61]: Thanks - learned something new! 🙂

**Review by tintoy — APPROVED** [review]

**Comment by tintoy** [comment]
Thanks!

File: docs/usage.rst (hunk 4 above)
Comment (piotrmaslanka) [reply docs/usage.rst]: It'd be a lie. It's not so much undefined as hard to put into words.

File: docs/usage.rst (hunk 3 above)
Comment (piotrmaslanka) [reply docs/usage.rst]: Yeah, fixed that.

**Pull request merged**